src/unknown_00BC20.c

```diff
@@ -1507,614 +1508,639 @@ void func_800235D0(s32 arg0) {
-            func_8004216C(arg0, arg1);
+        case 104:
+            obj_init_rangetrigger(obj, entry);
             break;
-        case 0x6B:
-        case 0x73:
-            func_80033F44(arg0, arg1);
+        case 107:
+        case 115:
+            obj_init_fireball_octoweapon(obj, entry);
             break;
-        case 0x6C:
-            func_80042210(arg0, arg1);
+        case 108:
+            obj_init_frog(obj, entry);
             break;
-        case 0x6D:
-            func_8003DBA0(arg0, arg1);
+        case 109:
+            obj_init_silvercoin_adv2(obj, entry);
             break;
-        case 0x74:
-            func_80042A1C(arg0, arg1);
+        case 116:
+            obj_init_levelname(obj, entry);
             break;
     }
 }
 
-#ifdef NON_EQUIVALENT
-s32 func_80023E30(s32 arg0) {
-    switch (arg0 - 1) {
-        case 0x00:
-            return 0x1F;
-        case 0x01:
-            return 0x13;
-        case 0x04:
-            return 0x16;
-        case 0x0B:
-            return 0x1B;
-        case 0x0D:
-        case 0x6E:
-            return 0x30;
-        case 0x10:
-        case 0x4C:
-            return 0x12;
-        case 0x45:
-        case 0x47:
-        case 0x5F:
-        case 0x60:
-            return 0x3B;
-        case 0x64:
-        case 0x65:
-        case 0x66:
-        case 0x67:
-            return 0x3A;
-        case 0x17:
-            return 0x04;
-        case 0x1E:
-            return 0x1B;
-        case 0x1F:
-        case 0x3F:
-        case 0x57:
-        case 0x6D:
-            return 0x12;
-        case 0x42:
-            return 0x30;
-        case 0x25:
-            return 0x39;
-        case 0x26:
-            return 0x12;
-        case 0x27:
-            return 0x09;
-        case 0x2C:
-            return 0x12;
-        case 0x2F:
-            return 0x09;
-        case 0x3E:
-            return 0x09;
-        case 0x31:
-        case 0x37:
-        case 0x4F:
-        case 0x53:
-        case 0x55:
-        case 0x72:
-            return 0x0B;
-        case 0x35:
-            return 0x0B;
-        case 0x49:
-        case 0x5E:
-        case 0x62:
-        case 0x63:
-            return 0x31;
-        case 0x5A:
-            return 0x01;
-        case 0x50:
-            return 0x0A;
-        case 0x06:
-        case 0x0C:
-        case 0x11:
-        case 0x19:
-        case 0x1D:
-        case 0x23:
-        case 0x28:
-        case 0x33:
-        case 0x36:
-        case 0x38:
-        case 0x43:
-        case 0x4D:
-        case 0x4E:
-        case 0x5C:
-        case 0x61:
-        case 0x6B:
-        case 0x73:
-            return 0x10;
-        case 0x51:
-            return 0x12;
-        case 0x30:
-        case 0x32:
-        case 0x3C:
-            return 0x02;
-        case 0x3D:
-            return 0x1B;
-        case 0x6C:
-            return 0x0B;
-        case 0x71:
-            return 0x01;
-        default:
-            return 0x00;
-    }
+s32 func_80023E30(s32 arg0){
```

Comment:
```suggestion
s32 func_80023E30(s32 arg0) {
```

Reply:
Fixed

Reply:
FYI you can commit suggested changes directly via the `Commit suggestion` button.